client/views/room/Header/QuickActions/QuickActions.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { memo, useContext, useCallback, useState, useEffect, useMemo } from 'react';
+import { BoxProps, ButtonGroup } from '@rocket.chat/fuselage';
+import { useMutableCallback } from '@rocket.chat/fuselage-hooks';
+import { Meteor } from 'meteor/meteor';
+import { FlowRouter } from 'meteor/kadira:flow-router';
+import { Session } from 'meteor/session';
+import toastr from 'toastr';
+
+import Header from '../../../../components/Header';
+import { useTranslation } from '../../../../contexts/TranslationContext';
+import { QuickActionsActionConfig, QuickActionsEnum } from '../../lib/QuickActions';
+import { useLayout } from '../../../../contexts/LayoutContext';
+import { useSetModal } from '../../../../contexts/ModalContext';
+import { QuickActionsContext } from '../../lib/QuickActions/QuickActionsContext';
+import ReturnChatQueueModal from '../../../../components/ReturnChatQueueModal';
+import ForwardChatModal from '../../../../components/ForwardChatModal';
+import TranscriptModal from '../../../../components/TranscriptModal';
+import CloseChatModal from '../../../../components/CloseChatModal';
+import { handleError } from '../../../../../app/utils/client';
+import { useEndpointAction } from '../../../../hooks/useEndpointAction';
+import { IRoom } from '../../../../../definition/IRoom';
+import { useAtLeastOnePermission, usePermission, useRole } from '../../../../contexts/AuthorizationContext';
+import { useUserId } from '../../../../contexts/UserContext';
+import { useOmnichannel } from '../../../../contexts/OmnichannelContext';
+
+const QuickActions = ({ room, className }: { room: IRoom; className: BoxProps['className'] }): JSX.Element => {
+	const setModal = useSetModal();
+	const { isMobile } = useLayout();
+	const t = useTranslation();
+	const { actions: mapActions } = useContext(QuickActionsContext);
+	const actions = (Array.from(mapActions.values()) as QuickActionsActionConfig[]).sort((a, b) => (a.order || 0) - (b.order || 0));
+	const visibleActions = isMobile ? [] : actions.slice(0, 6);
+	const [email, setEmail] = useState('');
+	const visitorRoomId = room.v?._id;
+	const rid = room._id;
+	const uid = useUserId();
+
+	const getVisitorInfo = useEndpointAction('GET', `livechat/visitors.info?visitorId=${ visitorRoomId }`);
+
+	const getVisitorEmail = useMutableCallback(async () => {
+		if (!visitorRoomId) { return; }
+		const { visitor: { visitorEmails } } = await getVisitorInfo();
+		setEmail(visitorEmails && visitorEmails.length > 0 && visitorEmails[0].address);
+	});
+
+	useEffect(() => {
+		getVisitorEmail();
+	}, [room, getVisitorEmail]);
+
+	const closeModal = useCallback(() => setModal(null), [setModal]);
+
+	const moveChat = (): void => Meteor.call('livechat:returnAsInquiry', rid, function(error: any) {
```

Comment:
quite sure you can use `useEndpointAction` we should not use `Meteor` deliberately inside react s context